packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/EdgeComponent.java

```diff
@@ -185,6 +196,92 @@ private List<Element> renderEdge(VariableManager variableManager, EdgeDescriptio
+            FadeDiagramElementEvent fadeDiagramElementEvent = (FadeDiagramElementEvent) optionalDiagramEvent.get();
+            if (fadeDiagramElementEvent.getElementIds().contains(id)) {
+                if (fadeDiagramElementEvent.fadeElement()) {
+                    modifiers.add(ViewModifier.Faded);
+                } else {
+                    modifiers.remove(ViewModifier.Faded);
+                }
+            }
+        }
+        return modifiers;
+    }
+
+    /**
+     * Compute the state of the current edge.
+     *
+     * This state is computed from the modifier set of the current edge and the states of the source and target nodes.
+     * If the new state of one of these nodes is {@link ViewModifier#Hidden}, the state of this new edge is
+     * {@link ViewModifier#Hidden} too. If these nodes are not hidden, the state of the current edge is the
+     * {@link ViewModifier#getDominantModifier(Collection) dominant state} of the set or
+     * {@link ViewModifier#DEFAULT_MODIFIER} if empty.
+     *
+     * @param optionalDiagramEvent
+     *            The optional diagram event, it is used to know the new state of the source and target nodes
+     * @param sourceNode
+     *            The source node element
+     * @param sourceId
+     *            The source node ID
+     * @param sourceTarget
+     *            The target node element
+     * @param targetId
+     *            The target node ID
+     * @param modifiers
+     *            The modifier set of the building edge
+     */
+    private ViewModifier computeState(Optional<IDiagramEvent> optionalDiagramEvent, Element sourceNode, String sourceId, Element targetNode, String targetId, Set<ViewModifier> modifiers) {
+
+        ViewModifier state = ViewModifier.getDominantModifier(modifiers).orElse(ViewModifier.DEFAULT_MODIFIER);
+
+        final ViewModifier sourceState = this.getStateFromElement(sourceNode);
+        final ViewModifier targetState = this.getStateFromElement(targetNode);
+
+        if (optionalDiagramEvent.isPresent() && optionalDiagramEvent.get() instanceof HideDiagramElementEvent) {
+            final var diagramEvent = (HideDiagramElementEvent) optionalDiagramEvent.get();
+            final boolean isSourceHidden = (diagramEvent.getElementIds().contains(sourceId) && diagramEvent.hideElement())
+                    || (!diagramEvent.getElementIds().contains(sourceId) && sourceState == ViewModifier.Hidden);
+
+            final boolean isTargetHidden = (diagramEvent.getElementIds().contains(targetId) && diagramEvent.hideElement())
+                    || (!diagramEvent.getElementIds().contains(targetId) && targetState == ViewModifier.Hidden);
+
+            if (isSourceHidden || isTargetHidden)
+                state = ViewModifier.Hidden;
+
+        } else {
+            if (sourceState == ViewModifier.Hidden || targetState == ViewModifier.Hidden)
+                state = ViewModifier.Hidden;
+        }
+        return state;
+    }
+
```

Comment:
Same comment regarding the use of final and the formatter. I don't think you have the right development environment.